kyo-core/native/src/main/scala/kyo/scheduler/IOPromisePlatformSpecific.scala

```diff
@@ -0,0 +1,18 @@
+package kyo.scheduler
+
+import kyo.Present
+import scala.scalanative.libc.stdatomic.AtomicRef
+import scala.scalanative.runtime.Intrinsics
+import scala.scalanative.runtime.fromRawPtr
+
+object IOPromisePlatformSpecific:
+
+    val stateHandle =
+        Present(
+            new IOPromise.StateHandle:
+                def compareAndSet[E, A](promise: IOPromise[E, A], curr: IOPromise.State[E, A], next: IOPromise.State[E, A]): Boolean =
+                    val ref = AtomicRef(fromRawPtr(Intrinsics.classFieldRawPtr(promise, "state")))
```

Comment:
Ideally, the field offset lookup should be separate like in the JVM version but I couldn't find an API for that. It seems ok since `AtomicRef` is an `AnyVal` and won't allocate.